components/resource_control/src/service.rs

```diff
@@ -0,0 +1,266 @@
+        pd_client: Arc<RpcClient>,
+    ) -> ResourceManagerService {
+        ResourceManagerService {
+            pd_client,
+            manager,
+            revision: 0,
+        }
+    }
+}
+
+impl ResourceManagerService {
+    pub async fn watch_resource_groups(&mut self) {
+        // Firstly, load all resource groups as of now.
+        match self.list_resource_groups() {
+            Ok((groups, revision)) => {
+                self.revision = revision;
+                for group in groups {
+                    self.manager.add_resource_group(group);
+                }
+            }
+            Err(e) => error!("failed to list resource groups, err: {:?}", e),
+        }
+        // Secondly, start watcher at loading revision.
+        loop {
+            match self
+                .pd_client
+                .watch_global_config(CONFIG_PATH.to_string(), self.revision)
+            {
+                Ok(mut stream) => {
+                    while let Some(grpc_response) = stream.next().await {
+                        match grpc_response {
+                            Ok(r) => {
+                                self.revision = r.get_revision();
+                                for item in r.get_changes() {
+                                    if let Ok(group) = protobuf::parse_from_bytes::<ResourceGroup>(
+                                        item.get_value().as_bytes(),
+                                    ) {
+                                        match item.get_kind() {
+                                            ItemKind::Put => self.manager.add_resource_group(group),
+                                            ItemKind::Delete => {
+                                                self.manager.remove_resource_group(item.get_name())
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                            Err(err) => {
+                                error!("failed to get stream, err: {:?}", err);
+                                std::thread::sleep(Duration::from_secs(1));
+                            }
+                        }
+                    }
+                }
+                Err(e) => {
+                    error!("failed to watch resource groups, err: {:?}", e);
+                    std::thread::sleep(Duration::from_secs(1));
+                }
+            }
+        }
+    }
+
+    fn list_resource_groups(&mut self) -> Result<(Vec<ResourceGroup>, i64)> {
+        match futures::executor::block_on(async move {
+            self.pd_client
+                .load_global_config(CONFIG_PATH.to_string())
+                .await
+        }) {
+            Ok((items, revision)) => {
+                let mut groups = Vec::default();
+                for item in items {
+                    if item.has_error() {
+                        error!(
```

Comment:
Why ignore this error? Pd should not return a item with error right?

Reply:
> Pd should not return a item with error right

 We will get error such as `GLOBAL_CONFIG_NOT_FOUND`or pd etcd get error
 I think the path does not exist or etcd get error are both very serious problem, Do you think we should panic at this point?

Reply:
We don't need to worry about loading null and we can remove `global config not found` error.